src/server/models/listItem/listItem.ts

```diff
@@ -0,0 +1,343 @@
+
+      ${pgescape(`WHERE "ListItem"."type" = %L`, list.type)}
+      AND ("ListItem"."jsonData"->'metadata'->>'emailVerified')::boolean
+      AND "Country".id = ${list.countryId}
+
+      ORDER BY "ListItem"."createdAt" DESC`);
+  } catch (error) {
+    logger.error(`approveLawyer Error ${error.message}`);
+    throw new Error("Failed to approve lawyer");
+  }
+}
+
+export async function findListItemById(
+  id: string | number
+): Promise<ListItemGetObject> {
+  try {
+    return (await prisma.listItem.findUnique({
+      where: { id: Number(id) },
+      include: {
+        address: {
+          select: {
+            id: true,
+            firstLine: true,
+            secondLine: true,
+            city: true,
+            postCode: true,
+            country: {
+              select: {
+                id: true,
+                name: true,
+              },
+            },
+          },
+        },
+      },
+    })) as ListItemGetObject;
+  } catch (error) {
+    logger.error(`findListItemById Error ${error.message}`);
+    throw new Error("Failed to approve lawyer");
+  }
+}
+
+export async function togglerListItemIsApproved({
+  id,
+  isApproved,
+  userId,
+}: {
+  id: number;
+  isApproved: boolean;
+  userId: User["id"];
+}): Promise<ListItem> {
+  const data: {
+    isApproved: boolean;
+    isPublished?: boolean;
+  } = { isApproved };
+
+  if (userId === undefined) {
+    throw new Error("togglerListItemIsApproved Error: userId is undefined");
+  }
+
+  if (!isApproved) {
+    data.isPublished = false;
+  }
+
+  try {
+    const [listItem] = await prisma.$transaction([
+      prisma.listItem.update({
+        where: { id },
+        data,
+      }),
+      recordListItemEvent({
+        eventName: isApproved ? "approve" : "disapprove",
+        itemId: id,
+        userId,
+      }),
+    ]);
+    return listItem;
+  } catch (error) {
+    logger.error(`togglerListItemIsApproved Error ${error.message}`);
+    throw error;
+  }
+}
+
+export async function togglerListItemIsPublished({
+  id,
+  isPublished,
+  userId,
+}: {
+  id: number;
+  isPublished: boolean;
+  userId: User["id"];
+}): Promise<ListItemWithAddressCountry> {
+  if (userId === undefined) {
+    throw new Error("togglerListItemIsPublished Error: userId is undefined");
+  }
+
+  try {
+    const [listItem] = await prisma.$transaction([
+      prisma.listItem.update({
+        where: { id },
+        data: { isPublished },
+        include: {
+          address: {
+            include: {
+              country: true,
+            },
+          },
+        },
+      }),
+      recordListItemEvent({
+        eventName: isPublished ? "publish" : "unpublish",
+        itemId: id,
+        userId,
+      }),
+    ]);
+
+    return listItem;
+  } catch (error) {
+    logger.error(`publishLawyer Error ${error.message}`);
+
+    throw new Error("Failed to publish lawyer");
+  }
+}
+
+export async function deleteListItem(
```

Comment:
Do we also need to cascade the deletion of the associated geo and address records?

Reply:
Opened a new issue for this